### PR TITLE
agent/httpserver: introduce simplified H1/H2C/H2/H3 server

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: agent/httpserver/listen.go [hunk 1]
```diff
@@ -10,39 +11,171 @@ import (
...
+func (*ServerListeners) checkCount(secure, quic, insecure int) error {
+	// same number of secure and quic listeners, and at least one
+	if secure == quic && secure > 0 {
+		// either no insecure listeners, or the same number as secure ones
+		if insecure == 0 || insecure == secure {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("inconsistent listeners count (secure:%v, quic:%v, insecure:%v)",
```
Comment (karasz) [inline agent/httpserver/listen.go:44]: In cases like this I would expect to get the why too, not only the what. Inconsistent listeners count... no many insecure listeners or insecure listeners cannot be in the same number as the secure ones.

**Review by karasz — APPROVED** [review]
LGTM

File: agent/httpserver/listen.go (hunk 1 above)
Comment (amery) [reply agent/httpserver/listen.go:44]: I'll improve this log entry merging `agent/httpserver` @karasz

**Comment by amery** [comment]
`shared/v0.4.0` tagged

**Comment by amery** [comment]
I'm stupid. `agent/v0.1.0` tagged

**Pull request merged**